polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -1029,3 +1033,125 @@ fn limit_and_sanitize_disputes<
+
+	if disabled_validators.len() == 0 {
+		// No disabled validators - nothing to do
+		return Ok(false)
+	}
+
+	// `BackedCandidate` contains `validator_indices` which are indecies within the validator group.
+	// To obtain `ValidatorIndex` from them we do the following steps:
+	// 1. Get `para_id` from `CandidateDescriptor`
+	// 2. Get the `core_idx` for the corresponding `para_id`
+	// 3. Get the validator group assigned to the corresponding core idx
+
+	// Map `para_id` to `core_idx` for step 2 from the above list
+	let para_id_to_core_idx = <scheduler::Pallet<T>>::scheduled_paras()
+		.map(|(core_idx, para_id)| (para_id, core_idx))
+		.collect::<BTreeMap<_, _>>();
+
+	// Flag which will be returned. Set to `true` if at least one vote is filtered.
+	let mut filtered = false;
+
+	// Process all backed candidates.
+	let backed_len_before = backed_candidates.len();
+	backed_candidates.retain_mut(|bc| {
+		// Get `core_idx` assigned to the `para_id` of the candidate (step 2)
+		let core_idx = match para_id_to_core_idx.get(&bc.descriptor().para_id) {
+			Some(core_idx) => *core_idx,
+			None => {
+				log::debug!(target: LOG_TARGET, "Can't get core idx got a backed candidate for para id {:?}. Dropping the candidate.", bc.descriptor().para_id);
+				return false
+			}
+		};
+
+		// Get relay parent block number of the candidate. We need this to get the group index
+		// assigned to this core at this block number
+		let relay_parent_block_number = match <shared::Pallet<T>>::allowed_relay_parents()
+			.acquire_info(bc.descriptor().relay_parent, None) {
+				Some((_, block_num)) => block_num,
+				None => {
+					log::debug!(target: LOG_TARGET, "Relay parent {:?} for candidate is not in the allowed relay parents. Dropping the candidate.", bc.descriptor().relay_parent);
+					return false
+				}
+			};
+
+
+		// Get the group index for the core
+		let group_idx = match <scheduler::Pallet<T>>::group_assigned_to_core(
+			core_idx,
+			relay_parent_block_number + One::one(),
+		) {
+			Some(group_idx) => group_idx,
+			None => {
+				log::debug!(target: LOG_TARGET, "Can't fetch group index for core idx {:?}. Dropping the candidate.", core_idx);
+				return false
+			},
+		};
+
+		// And finally get the validator group for this group index
+		let validator_group = match <scheduler::Pallet<T>>::group_validators(group_idx) {
+			Some(validator_group) => validator_group,
+			None => {
+				log::debug!(target: LOG_TARGET, "Can't get the validators from group {:?}. Dropping the candidate.", group_idx);
+				return false
+			}
+		};
+
+		// `validator_indices` in `BackedCandidate` are indecies within the validator group. Convert
+		// them to `ValidatorId`.
+		let voted_validator_ids = bc
+			.validator_indices
+			.iter()
+			.enumerate()
+			.filter_map(|(idx, bitval)| match *bitval {
+				true => validator_group.get(idx), // drop indecies not found in the validator group. This will lead to filtering the vote
+				false => None,
+			})
```

Comment:
clippy will complain about matching on bool instead of simple `if`

Reply:
This code is no longer needed.